agent_http.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+func startAgentListener(oa *OauthTokenHandler, agentConf *AgentConfig) {
+	http.HandleFunc("/oauth2/callback", func(w http.ResponseWriter, r *http.Request) {
+		token, err := oa.ExchangeToken(r.FormValue("code"))
+		if err != nil {
+			fmt.Fprintf(
+				w,
+				"error fetching token from web: %v",
+				err,
+			)
+			return
+		}
+		agentLeaseLoop(agentConf, token.IdToken)
+		fmt.Fprintf(w, "Auth is now complete and agent is running! you can close this window")
+	})
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		token, err := oa.getTokenFromFile()
+		if err != nil || token.IdToken == "" || token.Expiry.Before(time.Now()) {
+			log.Println("cannot get a valid cached token, need a new one")
+			// Get a url for the token challenge and redirect there
+			url, err := oa.prepareTokenWebChalenge()
+			if err != nil {
+				fmt.Fprintf(
+					w,
+					"error creating web url to get token: %v",
+					err,
+				)
+				return
+			}
+			http.Redirect(w, r, url, 302)
+			return
+		}
+		agentLeaseLoop(agentConf, token.IdToken)
+		fmt.Fprintf(w, "Agent refreshed and running! you can close this window now")
```

Comment:
Why is half the app using `log` and the other half is using `fmt`?

Reply:
I am using fmt when writing to `http.ResponseWriter` here, and log when the app is logging if I understand the question right? I updated everything that is an up log to use `log`

Reply:
Wrong place for a comment, but yes, everywhere else where we are writing to stdout

Reply:
Sorry to be picky, but if you are going to use punctuation then you need to capitalise the next word ))